presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -391,6 +394,9 @@ public boolean catalogExists(Session session, String catalogName)
 
         CatalogMetadata catalogMetadata = getCatalogMetadata(session, connectorId);
         ConnectorMetadata metadata = catalogMetadata.getMetadataFor(connectorId);
+
+        checkState(metadata.usesLegacyTableLayouts(), "getLayout() was called even though connector doesn't support legacy Table Layout");
```

Comment:
Well, that checkState depends on "metadata", which needs to be obtained from catalogMetadata, which needs the connectorId, so it's kind of obvious why it can't go at the beginning.